contrib/endpoints/src/api_manager/api_manager_impl.cc

```diff
@@ -24,25 +24,19 @@ using ::google::api_manager::proto::ServerConfig;
 namespace google {
 namespace api_manager {
 
-namespace {
-
-std::shared_ptr<ApiManager> CreateApiManager(
-    std::unique_ptr<ApiManagerEnvInterface> env,
-    std::unique_ptr<Config> config) {
-  return std::shared_ptr<ApiManager>(
-      new ApiManagerImpl(std::move(env), std::move(config)));
-}
-
-}  // namespace
-
 ApiManagerImpl::ApiManagerImpl(std::unique_ptr<ApiManagerEnvInterface> env,
-                               std::unique_ptr<Config> config)
-    : service_context_(
-          new context::ServiceContext(std::move(env), std::move(config))) {
+                               const std::string& server_config)
```

Comment:
You can have only one server_config.   It is suppose to override all service configs.